cbits/posix/find_executable.c

```diff
@@ -74,15 +104,36 @@ static char *get_executable_search_path(void) {
     return strdup(":");
 }
 
-/* Find the given executable in the executable search path. */
-char *find_executable(char *filename) {
-    /* If it's an absolute or relative path name, it's easy. */
-    if (strchr(filename, '/') && is_executable(filename)) {
+/* Find the given executable in the executable search path relative to
+ * workingDirectory (or the current directory, if NULL).
+ * N.B. the caller is responsible for free()ing the result.
+ */
+char *find_executable(char *working_dir, char *filename) {
+    /* Drop trailing slash from working directory if necessary */
+    if (working_dir) {
+        int workdir_len = strlen(working_dir);
+        if (working_dir[workdir_len] == '/') {
```

Comment:
I think this needs to check `working_dir[workdir_len - 1]`, right?

Reply:
Quite right!